charger/wattpilot.go

```diff
@@ -0,0 +1,145 @@
+package charger
+
+// LICENSE
+
+// Copyright (c) 2019-2022 andig
+
+// This module is NOT covered by the MIT license. All rights reserved.
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/evcc-io/evcc/api"
+	"github.com/evcc-io/evcc/util"
+	wattpilot "github.com/mabunixda/wattpilot"
+)
+
+// Wattpilot charger implementation
+type Wattpilot struct {
+	api *wattpilot.Wattpilot
+}
+
+func init() {
+	registry.Add("wattpilot", NewWattpilotFromConfig)
+}
+
+// NewWattpilotFromConfig creates a wattpilot charger from generic config
+func NewWattpilotFromConfig(other map[string]interface{}) (api.Charger, error) {
+	cc := struct {
+		URI      string
+		Password string
+		Cache    time.Duration
+	}{}
+	if err := util.DecodeOther(other, &cc); err != nil {
+		return nil, err
+	}
+
+	if cc.URI == "" || cc.Password == "" {
+		return nil, errors.New("must have one  uri and password")
+	}
+
+	return NewWattpilot(cc.URI, cc.Password, cc.Cache)
+}
+
+// NewWattpilot creates Wattpilot charger
+func NewWattpilot(uri, password string, cache time.Duration) (api.Charger, error) {
+	log := util.NewLogger("wattpilot")
+	c := &Wattpilot{}
+
+	log.INFO.Println("Connecting wattpilot local", uri)
```

Comment:
```suggestion
	log.INFO.Println("Connecting to wattpilot at %s", uri)
```

Reply:
I‘ll remove the logging as it doesn‘t add any information over the config. 

@mabunixda how does the library handle logging?

Reply:
At the moment there is no output when using the library, with 1 exception on function `StatusInfo` that is used on the wattpilot-shell implementation.